### DISCUSSION TITLE
Altitude in Charts signficantly different to what's in the recording

### DISCUSSION THREAD

**Discussion by zell-mbc** [OpenTracksApp/OpenTracks#1631] [Q&A] [answered]
I have seen the many questions related to altitude/elevation here, but I still got a question, or maybe looking for confirmation.

My recordings don't show any value in the intervals tab, which is due to the fact that my phone (OnePlus 8 Pro) doesn't have a barometer, understood.
The "By time" and "By distance" tabs do show altitude, which as I understand is calculated based on GPS data.

Now, I just completed a bike trip, with OpenTracks, OSMand~, and Withings running in parallel.
It turns out that Opentracks is off by ~50m, max elevation with OpenTracks shows as ~600m while it should be 650m. The others all show the same values as what OpenStreetMap reports for individual points, 650m for my max today.
![image](https://github.com/OpenTracksApp/OpenTracks/assets/29018767/84151a9f-6b2d-455e-ac0c-1e9558856a3f)

However, when I check the file recorded by OpenTracks with a text editor I can see that there are a few <ele>650.1</ele> records? And when I open this recording in e.g. https://opentopomap.org, I do get the correct elevation values as well? It looks like somehow what's shown in the OpenTracks charts is different from what's in the actual recording?
![image](https://github.com/OpenTracksApp/OpenTracks/assets/29018767/306ee55b-a8ef-46cd-9c36-ab8d2e828424)

Given that OpenTracks doesn't have the map available during recording, where are the correct values coming from? And if the correct values are somehow available, why bother showing the calculated GPS values in the charts?

Aside from OpenTracks's simplicity, the fact that I can automatically store my tracks on Nextcloud is the killer feature for me. Which means, it doesn't matter if OpenTracks Altitude Charts are  off, because I will do my reviews/sharing/etc. in a different app anyway.

**Comment by dennisguse** [comment]
OpenTracks UI shows altitude as EGM2008 while GPS (and also GPX/KML) only provide altitude as WGS84. 
However, WGS84 has quite a difference to the actual sea level and EGM2008 is way closer. 
OpenTracks comes with data bundles for thiw correction and for Android 14, this became a native API :)

**Comment by zell-mbc** [comment]
Thanks for getting back Dennis, does that mean you calculate two values in parallel? One is shown in the charts and the other one goes into the GPX/KML file?
And the other apps I tried today probably use WGS84 only, maybe even use the same library, which is why they arrive at roughly the same values as the OpenTracks GPX/KML file.
Also sounds like what goes into the GPX file isn't really under your control, probably because GPX/KML data is created by an external library? Otherwise you could add the more accurate WGS84 elevation values to the GPX/KML file I guess?

I think I need to find a few spots with a known and officially published elevation to check how much things are off, just out of curiosity. But in any case, OpenTracks GPX/KML is as good or bad as all the others which works for me.

Also didn't realize that elevation isn't part of the OSM, https://help.openstreetmap.org/questions/3069/elevation-maps

Edit: Found one: 
![image](https://github.com/OpenTracksApp/OpenTracks/assets/29018767/ffc55583-43e9-4af3-a706-de86a9fc4248)

…and can also confirm that WGS84 is a lot more accurate. According to Wikipedia this peak is 766m, I had Komoot and OpenTracks running to test things out when I got up a while ago. Komoot is off by ~50m, shows 800m as max, and so is the the OpenTracks GPX file I captured. The OpenTracks chart though looks pretty close. 

Now I need a calibration setting for GPX/KML…or call it a day and ignore elevation :-)

**Reply by dennisguse** [reply, marked as answer]
The other way around. 


We get WGS84 from GPS (it is the default for almost everything) and it has quite a difference. And then we use data from geographiclib (aka an approximation of the EGM2008 altitude correction function) to adjust the shown altitude in OpenTracks UI (not the export). 
Almost all applications can only show WGS84. 

There are some issues describing this problem in more detail in the repo. 

PS With EGM2008 correction (using the bundled 5° data) the difference between MSL and shown altitude is about 1.5m if I recall correctly. WGS84 is up to 50m off. 

And there are GPS calibration points. I was at one in Berlin (close to main station) to verify the whole thing.

**Reply by zell-mbc** [reply]
I see, thanks again for clarifying.
OSMand seems to offer a correction map,  "World altitude correction" , https://groups.google.com/g/osmand/c/9bRqJlSQrCk
I'll give this a try and if the results are any better I will report back here. Maybe this could be an enhancement, or maybe this is the same as the geographiclib you mention above.

**Reply by zell-mbc** [reply]
Right, 50km later I have new data.

Went to a lake nearby which according to Wikipedia is on 578,3m.
OSMand with "World altitude correction" shows 578,9 m
OpenTracks reports 626,5m
Seems like this correction map does add value

Maybe an enhancement to consider…

**Reply by dennisguse** [reply]
Would recommend to check out a GPS reference location: https://www.lgl-bw.de/unsere-themen/Geoinformation/Kontrollpunkte/

**Reply by zell-mbc** [reply]
Very cool! One's even in my home town, I had no idea :-)
